include/deal.II/numerics/derivative_approximation.h

```diff
@@ -173,29 +173,23 @@ namespace DerivativeApproximation
    * In a parallel computation the @p solution vector needs to contain the
    * locally relevant unknowns.
    */
-  template <int dim,
-            template <int, int> class DoFHandlerType,
-            class InputVector,
-            int spacedim>
+  template <int dim, class InputVector, int spacedim>
   void
-  approximate_gradient(const Mapping<dim, spacedim> &       mapping,
-                       const DoFHandlerType<dim, spacedim> &dof,
-                       const InputVector &                  solution,
-                       Vector<float> &                      derivative_norm,
-                       const unsigned int                   component = 0);
+  approximate_gradient(const Mapping<dim, spacedim> &   mapping,
+                       const DoFHandler<dim, spacedim> &dof,
```

Comment:
The tester complains regarding the all-headers test on `derivative_approximation.h` that we should either forward-declare `DoFHandler` or include `dof_handler.h`, probably the latter.